www/activities/foods-meals-recipes/js/food-editor.js

```diff
@@ -17,6 +17,35 @@
   along with app.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+
+
+function evalInpEquation(strIn)
+{
+  let ret = 0;
+  let strInTmp = strIn.trim();
+
+  if (strInTmp.length == 0 || strInTmp == "+" || strInTmp == "-")
+    return NaN;
+
+  let cleanStr = strInTmp.replaceAll("+", " + ").replaceAll("-", " - ").replaceAll("  ", " ");
+  let tokens = cleanStr.split(" ");
+
+  if (tokens[0] == "" && (tokens[1] == "+" || tokens[1] == "-"))
+      tokens[0] = "0";
+
+  ret = parseInt(tokens[0]);
+  for (let i = 1; i < (tokens.length - 1); i+=2)
+  {
+    if (tokens[i] == "+")
+        ret += (parseInt(tokens[i+1]) | 0);
+    else if (tokens[i] == "-")
+        ret -= (parseInt(tokens[i+1]) | 0);
```

Comment:
I think you meant to use the logical or operator `||` instead of the bitwise or operator `|` here.